src/interface/report/CombatantInfoFaker.ts

```diff
@@ -1,14 +1,162 @@
 import SPECS from 'game/SPECS';
 import { specificGearSets } from 'interface/report/CombatantInfoFakerGearsets';
+import COVENANTS from 'game/shadowlands/COVENANTS';
+import SOULBINDS from 'game/shadowlands/SOULBINDS';
+import { Conduit } from 'parser/core/Events';
+
+const debugGear = false;
+const debugCovenant = false;
+const debugSoulbind = false;
 
 export function generateFakeCombatantInfo(player: any) {
   const fakedPlayer = player;
   fakedPlayer.gear = fakeGearGenerator(player.specID);
+  fakedPlayer.gear = fakeLegendaryGeneration(player.gear);
   fakedPlayer.auras = fakeBuffGenerator();
+  fakedPlayer.covenant = fakeCovenantGenerator(player.specID);
+  fakedPlayer.soulbind = fakeSoulbindGenerator(player.specID);
+  fakedPlayer.conduits = fakeConduitGenerator();
   fakedPlayer.error = null;
   return fakedPlayer;
 }
 
+//region Shadowlands Systems
+
+//region Covenant Generation
+
+function fakeCovenantGenerator(specID: number) {
```

Comment:
The design of this is a little weird, I would have thought doing something like this would be easier for people to use:

```typescript

type FakeInfo = 
  {
    covenant: Covenant, // or id?
    soulbind: Soulbind // or id?
    ... conduits, etc
  }

const SPEC_CONFIGS: { [specId: number]: FakeInfo } = 
{
  [SPECS.MARKSMANSHIP_HUNTER.id]: { covenant: COVENANTS.VENTHYR, soulbind: SOULBINDS.GENERAL_DRAVEN, [...] } 
}
```

So that people could change their specs all in one place instead of having to go through each of the arrays. Like every other comment this one is pretty stylistic and not blocking

Reply:
That is way easier 😊